File: JavaScript/CompareDates.js
/*CompareDates.js
*
*   Using dates in JavaScript and comparing them.
*   LU: 10/04/15
*/

function checkCoupon(enteredCode, correctCode, currentDate, expirationDate){
  
  if(correctCode === enteredCode)
  {
    // Convert currentDate and expirationDate to appropriate format
    var intCurrentDate = parseInt(Date.parse(currentDate));
    var intExpireDate = parseInt(Date.parse(expirationDate));
    console.log(intCurrentDate);
    console.log(intExpireDate);
    
    // Compare dates
    if(intCurrentDate <= intExpireDate)
      return true;
    
    return false;
  }
  else
    return false;
    
  // Code should never reach here.
  return false;
}

File: C Sharp/CompareDates.cs
/*CompareDates.cs
*
*	C# example on how to use DateTime.
*	LU: 10/04/15
*/

using System;

public static class Kata
{
  public static bool CheckCoupon(string enteredCode, string correctCode, string currentDate, string expirationDate)
  {
    int date_comparison;
  	DateTime current_dt = Convert.ToDateTime(currentDate);
    DateTime expire_dt = Convert.ToDateTime(expirationDate);
    //Console.WriteLine(dt);
        
    // Compare the dates
	date_comparison = DateTime.Compare(current_dt, expire_dt);	
    if(date_comparison <= 0)
    {
    	if(enteredCode == correctCode)
      	return true;
    }
    else
    	return false;
    
    // Should never reach here.
    Console.WriteLine("Something strange happened. Retry please.")
    return false;
  }
}